numpy/core/src/multiarray/ctors.c

```diff
@@ -3049,12 +3066,20 @@ _calc_length(PyObject *start, PyObject *stop, PyObject *step, PyObject **next, i
         }
         return -1;
```

Comment:
Need to `DECREF` `zero` here too - or better yet, just allocate it further down.